### PR TITLE
Issue #19

### PR DESCRIPTION
**Pull request by Lamaosman2012**
Waiting for review

### CODE REVIEW

File: class_social/posts.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        posts_list = db.load_posts()
+
+        for post in posts_list:
+
+            if post.creator.role == role:
+
+                 result_post.append(post)
+
+        return result_post
+
```
Comment (mathiasbrito-dci) [inline class_social/posts.py:23]: Posts by creator role never return None, and this is what is being tested, the mock is returning None.

**Review by mathiasbrito-dci — CHANGES_REQUESTED** [review]
Some minor issues, please adjust.

File: class_social/posts.py (hunk 1 above)
Comment (divyaChandran10) [reply class_social/posts.py:23]: Yes, it doesn't return None but returns empty list if no posts are available from the given role. We changed the name of the test_case. please find the code below:

def test_get_posts_by_creator_role_must_return_empty_list_if_no_post_is_found_with_the_specified_role():
    with patch('class_social.db.load_posts') as mocked_load_posts:
        mocked_load_posts.return_value = []
        controller = PostController()
        result = controller.get_posts_by_creator_role('non_existent_role')
        print(result)

        assert result == []

Comment (mathiasbrito-dci) [reply class_social/posts.py:23]: ok, good...

File: class_social/models.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ class User(BaseModel):
     website: Optional[str]
     connections: Optional[List['User']]
     about: Optional[str]
+    role: str
```
[the remark this answers is not in the document]
Comment (mathiasbrito-dci) [reply class_social/models.py:20]: ok, after that I think the code is ready to be merged...

**Review by mathiasbrito-dci — APPROVED** [review]

**Pull request merged**